crates/wascc-provider/src/lib.rs

```diff
@@ -214,6 +221,38 @@ impl<S: Store + Send + Sync> Provider for WasccProvider<S> {
         let client = kube::Client::new(self.kubeconfig.clone());
         let volumes = Ref::volumes_from_pod(&self.volume_path, &pod, &client).await?;
         for container in pod.containers() {
+            let mut port_assigned: i32 = 0;
+            info!(
+                "The following ports are already taken: {:?}",
+                self.port_set.lock().unwrap().keys()
```

Comment:
Because this is a locking operation (that involves some overhead and locks other threads from using it). This should probably be a `trace!`.

Reply:
should we assign the lock to another variable? I see that we're acquiring a lock on this same variable on lines 234 and 236, so perhaps it may be easier to block `port_set` for the entire function call?